File: backend/src/main/java/de/ckollmeier/burgerexpress/backend/dto/MenuInputDTO.java
package de.ckollmeier.burgerexpress.backend.dto;

import de.ckollmeier.burgerexpress.backend.interfaces.NamedDTO;
import de.ckollmeier.burgerexpress.backend.interfaces.PricedDTO;

import java.util.List;
import java.util.Map;

public record MenuInputDTO(
        String name,
        String price,
        List<String> dishIds,
        Map<String, AdditionalInformationDTO> additionalInformation
) implements NamedDTO, PricedDTO {
}

File: frontend/src/types/MenuInputDTO.ts
import { type AdditionalInformationInputDTO } from "./AdditionalInformationInputDTO.ts";

export type MenuInputDTO = {
    name: string;
    price: string;
    dishIds: string[];
    additionalInformation: Record<string, AdditionalInformationInputDTO>;
};

export type MenuInputDTOWithId = MenuInputDTO & {
    id?: string;
};